### PR TITLE
GAWB-2456: return compatible configs for a method snapshot

### PR DESCRIPTION
**Pull request by davidangb**
New Agora endpoint for compatible configurations.

Requires the orch passthrough at broadinstitute/firecloud-orchestration#505

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/resources/swagger/agora.yaml [hunk 1]
```diff
@@ -363,6 +363,54 @@ paths:
...
+        default:
+          description: Unexpected error
+          schema:
+            type: object
+            $ref: '#/definitions/Error'
+      security:
+        - authorization:
+            - openid
+            - email
+            - profile
```
Comment (rushtong) [inline src/main/resources/swagger/agora.yaml]: Nit: not necessarily your code, but the security block could probably be refactored to the top level.

File: src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala [hunk 2]
```diff
@@ -369,6 +369,44 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
...
+          }
+        }
+      }
+    }
+  }
+
+  // copied from org.broadinstitute.dsde.rawls.jobexec.MethodConfigResolver
+  def parseWDL(wdl: String): Try[wdl4s.Workflow] = {
+    val parsed: Try[WdlNamespaceWithWorkflow] = WdlNamespaceWithWorkflow.load(wdl, Seq()).recoverWith { case t: SyntaxError =>
+      Failure(new AgoraException("Failed to parse WDL: " + t.getMessage()))
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala]: unnecessary `new` in `new AgoraException`
`getMessage()` can just be `getMessage`

**Review by rushtong — APPROVED** [review]
👍 LG, works as expected. Minor comments inline that do not need a re-review if changed.

File: src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala (hunk 2 above)
Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala]: good catch on those; as the the comment says this was copy-paste so I didn't even check that.

**Pull request merged**